livekit-rtc/livekit/rtc/room.py

```diff
@@ -265,35 +269,40 @@ def _on_room_event(self, event: proto_room.RoomEvent):
             )
             self.emit("participant_connected", rparticipant)
         elif which == "participant_disconnected":
-            sid = event.participant_disconnected.participant_sid
-            rparticipant = self.participants.pop(sid)
-            self.participants_by_identity.pop(rparticipant.identity)
+            identity = event.participant_disconnected.participant_identity
+            rparticipant = self.remote_participants.pop(identity)
             self.emit("participant_disconnected", rparticipant)
         elif which == "local_track_published":
             sid = event.local_track_published.track_sid
-            lpublication = self.local_participant.tracks[sid]
+            lpublication = self.local_participant.track_publications[sid]
             track = lpublication.track
             self.emit("local_track_published", lpublication, track)
         elif which == "local_track_unpublished":
             sid = event.local_track_unpublished.publication_sid
-            lpublication = self.local_participant.tracks[sid]
+            lpublication = self.local_participant.track_publications[sid]
             self.emit("local_track_unpublished", lpublication)
         elif which == "track_published":
-            rparticipant = self.participants[event.track_published.participant_sid]
+            rparticipant = self.remote_participants[
+                event.track_published.participant_identity
+            ]
             rpublication = RemoteTrackPublication(event.track_published.publication)
-            rparticipant.tracks[rpublication.sid] = rpublication
+            rparticipant.track_publications[rpublication.sid] = rpublication
             self.emit("track_published", rpublication, rparticipant)
         elif which == "track_unpublished":
-            rparticipant = self.participants[event.track_unpublished.participant_sid]
-            rpublication = rparticipant.tracks.pop(
+            rparticipant = self.remote_participants[
```

Comment:
nit: I think we should always use .get() on these to guard against the object not being present. if participant is removed, it'd be better to handle the None responses.

Reply:
@theomonnom is it possible for track_unpublished (and friends) to come after the participant has been disconnected. I'm not super familiar with the async rust stuff and what guarantees we have for ffi events.